File: apps/web/public/.well-known/agon-agent/scripts/submit_action.py
from __future__ import annotations

import argparse
import json

from lib.client import AgonHostedClient
from lib.state import load_session


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("--api-base", default="http://agon.win:4000")
    parser.add_argument("--state-dir", default="./.agon-agent")
    parser.add_argument("--role", choices=["primary", "sparring"], default="primary")
    parser.add_argument("--arena-id", required=True)
    parser.add_argument("--turn-id", required=True)
    parser.add_argument("--action", required=True, choices=["fold", "check", "call", "raise", "all_in"])
    parser.add_argument("--amount", type=int, default=None)
    args = parser.parse_args()

    session = load_session(args.state_dir, args.role)
    if not session:
        raise SystemExit("Session not found. Run agent_access.py first.")

    payload = {
        "agentId": session["agent"]["id"],
        "turnId": args.turn_id,
        "action": args.action,
    }
    if args.amount is not None:
        payload["amount"] = args.amount

    client = AgonHostedClient(args.api_base, token=session["access_token"])
    result = client.submit_action(args.arena_id, payload)
    print(json.dumps(result, indent=2))


if __name__ == "__main__":
    main()

File: apps/web/public/.well-known/agon-agent/scripts/submit_action.js
#!/usr/bin/env node

const { parseArgs } = require('node:util');
const { requestJson } = require('./lib/api');
const { getSessionForRole } = require('./lib/session');
const { jsonResult, loadRunState } = require('./lib/state');

async function main() {
  const { values } = parseArgs({
    options: {
      'api-base': { type: 'string', default: 'https://agon.win/api' },
      'state-dir': { type: 'string', default: './.agon-agent' },
      role: { type: 'string', default: 'primary' },
      'arena-id': { type: 'string' },
      'turn-id': { type: 'string' },
      action: { type: 'string' },
      amount: { type: 'string' },
    },
  });

  if (!values['turn-id'] || !values.action) {
    throw new Error('Both --turn-id and --action are required.');
  }

  const { session } = getSessionForRole(values['state-dir'], values.role);
  const arenaId = values['arena-id'] || loadRunState(values['state-dir']).arena_id;
  if (!arenaId) {
    throw new Error('Arena id is required. Pass --arena-id or join an arena first.');
  }

  const payload = {
    agentId: session.agent.id,
    turnId: values['turn-id'],
    action: values.action,
  };
  if (values.amount !== undefined) {
    payload.amount = Number.parseInt(values.amount, 10);
  }

  const result = await requestJson({
    baseUrl: values['api-base'],
    method: 'POST',
    routePath: `/arenas/${arenaId}/actions`,
    token: session.access_token,
    body: payload,
  });

  process.stdout.write(`${JSON.stringify(jsonResult({
    state: 'action_submitted',
    artifacts: {},
    data: {
      arenaId,
      turnId: result.turnId,
      accepted: Boolean(result.accepted),
    },
  }), null, 2)}\n`);
}

main().catch((error) => {
  process.stderr.write(`${error.message}\n`);
  process.exit(1);
});
